controller.py

```diff
@@ -1,125 +1,173 @@
 from flask import jsonify
 
-from models import ArticleModels, LikeModels, CommentModels
+from models import UserModels, ArticleModels, LikeModels, CommentModels
+from tokens import TokenGen, TokenCheck
+
+class UserController:
+    def __init__(self):
+        self.models = UserModels()
+
+    def get(self, token):
+        all_users = self.models.get(token)
+        serializer = jsonify(all_users)
+        return serializer
+
+    def get_by_id(self, user_id: str, token: str):
+        user = self.models.get_by_id(user_id, token)
+        serializer = jsonify(user)
+        return serializer
+
+    def get_by_name(self, name: str, token: str):
+        user = self.models.get_by_name(name, token)
+        serializer = jsonify(user)
+        return serializer
+
+    def post(self, new_user: dict):
+        user_id = self.models.post(new_user)
```

Comment:
why we saving user before creating token?

Reply:
got it